app/controllers/pages_controller.rb

```diff
@@ -26,18 +26,20 @@ def become_a_taskee
   end
 
   def search
-    @taskees = get_taskees_by_search
+    if params[:searcher] || session[:searcher]
+      session[:searcher] = params[:searcher] if params[:searcher]
+      @taskees = get_taskees_by_search(session[:searcher])
+    end
     @taskees = nil if @taskees == []
```

Comment:
It is needed because there is a check for the value of @taskee in the view. It could either be nil or an array. So the empty array would throw an error. Moreover, I think I should have returned an empty array in the `get_taskees_by_search` method because PG will return a PG object if it doesn't find any record instead of an empty array. So forcing the method to return an empty array could be better for manipulation.